Documentation/ceph-recover-pod-on-LostNode.md

```diff
@@ -0,0 +1,138 @@
+
+```console
+kubectl -n rook-ceph get pod
+```
+>```
+>csi-rbd-demo-depl-69cd495f9b-8ldf9                  0/1     ContainerCreating   0          24m
+>csi-rbd-demo-depl-69cd495f9b-nl6hf                  1/1     Terminating         0          32m
+>```
+
+And, when you describe the pod stuck in `ContainerCreating` you see `Warning` like below
+
+```console
+$ kubectl -n rook-ceph describe pod csi-rbd-demo-depl-69cd495f9b-8ldf9
+
+Warning  FailedAttachVolume  17m                  attachdetach-controller                Multi-Attach error for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a" Volume is already used by pod(s) csi-rbd-demo-depl-69cd495f9b-rwdxq
+Warning  FailedMount         3m43s (x6 over 15m)  kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[mypvc default-token-s2fdk]: timed out waiting for the condition
+```
+
+**Solution:**
+
+Delete the pod stuck in `Terminating` state forcefully
+```console
+$ kubectl -n rook-ceph delete pod csi-rbd-demo-depl-69cd495f9b-nl6hf --grace-period 0 --force
+```
+
+**And** then wait for few minutes.
+
+To verify
+
+```console
+kubectl -n rook-ceph get pod
+```
+>```
+> csi-rbd-demo-depl-69cd495f9b-8ldf9                  1/1     Running       0          39m
+>```
+
+And, when you describe the pod now
+
+```console
+$ kubectl -n rook-ceph delete pod csi-rbd-demo-depl-69cd495f9b-nl6hf
+
+Multi-Attach error for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a" Volume is already used by pod(s) csi-rbd-demo-depl-69cd495f9b-rwdxq
+Warning  FailedMount             6m6s               kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[default-token-s2fdk mypvc]: timed out waiting for the condition
+Normal   SuccessfulAttachVolume  3m6s               attachdetach-controller                AttachVolume.Attach succeeded for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a"
+Warning  FailedMount             94s (x8 over 19m)  kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[mypvc default-token-s2fdk]: timed out waiting for the condition
+Normal   AddedInterface          78s                multus                                 Add eth0 [10.129.2.21/23]
+Normal   Pulling                 78s                kubelet, ip-10-0-159-129.ec2.internal  Pulling image "docker.io/library/nginx:latest"
+Normal   Pulled                  74s                kubelet, ip-10-0-159-129.ec2.internal  Successfully pulled image "docker.io/library/nginx:latest"
+Normal   Created                 74s                kubelet, ip-10-0-159-129.ec2.internal  Created container web-server
+Normal   Started                 74s                kubelet, ip-10-0-159-129.ec2.internal  Started container web-server
+
+```
+
+# Node lost and still has the device in it
```

Comment:
How is this case different from the first one? The terminating pod was no longer found?

Reply:
when a node is lost actual sometimes a two-step process is required to recover the application

* Removing the volume attachment
* Removing the watchers on the image

Reply:
Why wouldn't those steps always be required? It seems like they only wouldn't apply if the node is still online for cleanup, and that wouldn't be a lost node scenario.